src/auxinfo/proof.rs

```diff
@@ -38,34 +41,35 @@ impl AuxInfoProof {
         let auxinfo_proof: AuxInfoProof = deserialize!(&message.unverified_bytes)?;
         Ok(auxinfo_proof)
     }
+
+    /// Construct a proof that the modulus `N` is a valid product of two large
+    /// primes `p` and `q` (`𝚷[mod]`) and that neither `p` nor `q` are small
+    /// (`𝚷[fac]`).
+    ///
+    /// Note: The [`VerifiedRingPedersen`] argument **must be** provided by the
+    /// verifier!
     #[allow(clippy::too_many_arguments)]
     pub(crate) fn prove<R: RngCore + CryptoRng>(
         rng: &mut R,
         context: &<AuxInfoParticipant as InnerProtocolParticipant>::Context,
         sid: Identifier,
         rho: [u8; 32],
-        setup_params: &VerifiedRingPedersen,
+        verifier_params: &VerifiedRingPedersen,
         N: &BigNumber,
         p: &BigNumber,
         q: &BigNumber,
     ) -> Result<Self> {
-        let mut pimod_transcript = Transcript::new(b"PaillierBlumModulusProof");
-        pimod_transcript.append_message(b"PiMod ProofContext", context.as_bytes());
-        pimod_transcript.append_message(b"Session Id", &serialize!(&sid)?);
-        pimod_transcript.append_message(b"rho", &rho);
+        let mut pimod_transcript = Self::pimod_transcript(context, sid, rho)?;
         let pimod = PiModProof::prove(
             &PiModInput::new(N),
             &PiModSecret::new(p, q),
             context,
             &mut pimod_transcript,
             rng,
         )?;
-        let mut pifac_transcript = Transcript::new(b"PiFacProof");
-        pifac_transcript.append_message(b"PiFac ProofContext", context.as_bytes());
-        pifac_transcript.append_message(b"Session Id", &serialize!(&sid)?);
-        pifac_transcript.append_message(b"rho", &rho);
+        let mut pifac_transcript = Self::pifac_transcript(context, sid, rho)?;
```

Comment:
I think this should be appended on top of the transcript for pimod, actually. You can do that here or make a separate issue (I'm sensing an "improve code quality in AuxInfoProof" ticket)

Reply:
Not sure I follow. Are you saying we should have a single transcript for both proofs?

Reply:
Yes.

Reply:
E.g. the new `pifac_transcript` method should take a transcript as a parameter and update it, rather than making a new one from scratch.

Reply:
Done (I think, please take a look :-) )

Reply:
Yes, perfect. Thank you!